### PR TITLE
467 create a class for bounding rectangle

### PR DESCRIPTION
**Pull request by GabrielJMS**
* **Please check if the PR fulfills these requirements**
- [x] The commit messages follows our guidelines (explicit, in english)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] CHANGELOG.md has been updated

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
- [ ] Bug fix:
- [x] New features:
- [ ] Documentation
- [ ] Optimization

* **Other information**:

### CODE REVIEW

File: volmdlr/core.py [hunk 1]
```diff
@@ -794,36 +929,36 @@ def distance_to_bbox(self, bbox2):
...
+        elif self.xmax < point[0]:
+            dx = point[0] - self.xmax
         else:
-            if point[0] < self.xmin:
-                dx = self.xmin - point[0]
-            elif self.xmax < point[0]:
-                dx = point[0] - self.xmax
-            else:
-                dx = 0
+            dx = 0
```
Comment (AxelRinghausen) [inline volmdlr/core.py:947]: `dx = abs(self.xmin - point[0]`

File: volmdlr/core.py [hunk 2]
```diff
@@ -794,36 +929,36 @@ def distance_to_bbox(self, bbox2):
...
-            elif self.ymax < point[1]:
-                dy = point[1] - self.ymax
-            else:
-                dy = 0
+        if point[1] < self.ymin:
+            dy = self.ymin - point[1]
+        elif self.ymax < point[1]:
+            dy = point[1] - self.ymax
+        else:
+            dy = 0
```
Comment (AxelRinghausen) [inline volmdlr/core.py:954]: `dy = abs(self.ymin - point[1])`

File: volmdlr/core.py [hunk 3]
```diff
@@ -794,36 +929,36 @@ def distance_to_bbox(self, bbox2):
...
-            elif self.zmax < point[2]:
-                dz = point[2] - self.zmax
-            else:
-                dz = 0
+        if point[2] < self.zmin:
+            dz = self.zmin - point[2]
+        elif self.zmax < point[2]:
+            dz = point[2] - self.zmax
+        else:
+            dz = 0
```
Comment (AxelRinghausen) [inline volmdlr/core.py:961]: `dz = abs(self.zmin - point[2])`

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]
Check the "good practices" section in Notion for better typing and docstrings.

**Comment by masfaraud** [comment]
Ok great now use it as replacement of implicit xmin,xmax, ymin, ymax in 
image

**Review by masfaraud — DISMISSED** [review]

**Review by WirajanDASILVA — DISMISSED** [review]

File: volmdlr/wires.py [hunk 4]
```diff
@@ -4739,8 +4739,8 @@ def sewing_with(self, other_poly3d, x, y, normal, resolution=20):
 
         bbox_self2d, bbox_other2d = self_poly2d.bounding_rectangle(), \
             other_poly2d.bounding_rectangle()
-        position = [abs(value) for value in bbox_self2d] \
-            + [abs(value) for value in bbox_other2d]
+        position = [abs(value) for value in bbox_self2d.items()] \
+            + [abs(value) for value in bbox_other2d.items()]
```
Comment (AxelRinghausen) [inline volmdlr/wires.py:4743]: `bbox_self2d` and `bbox_other2d` are not dictionaries.

**Review by AxelRinghausen — CHANGES_REQUESTED** [review]

File: volmdlr/wires.py (hunk 4 above)
Comment (GabrielJMS) [reply volmdlr/wires.py]: Indeed. bbox_self2d and bbox_other2d should receive the bounds of the BoundingRectangle. I had created a method called items() for BoundingRectangle but then I changed the method name to bounds(). Thank you

**Review by AxelRinghausen — APPROVED** [review]

**Pull request merged**